sources/api/apiclient/src/network.rs

```diff
@@ -0,0 +1,221 @@
+use std::path::Path;
+use tokio::io::AsyncReadExt;
+
+/// Configures network settings from the given input source (URI or stdin), writing the content
+/// directly to /.bottlerocket/net.toml via the API server.
+pub async fn configure<P>(socket_path: P, input_source: String) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    // Get the network configuration content from the input source
+    let content = get_content(&input_source).await?;
+
+    // Send the content to the API server to write to /.bottlerocket/net.toml
+    let uri = "/actions/network/configure";
+    let method = "POST";
+    let (_status, _body) = crate::raw_request(&socket_path, uri, method, Some(content))
+        .await
+        .context(error::ConfigureRequestSnafu {
+            input_source,
+            uri,
+            method,
+        })?;
+
+    Ok(())
+}
+
+/// Retrieves the network configuration content from the given source location.
+async fn get_content<S>(input_source: S) -> Result<String>
+where
+    S: Into<String>,
+{
+    let input_source = input_source.into();
+
+    // Read from stdin if "-" was given.
+    if input_source == "-" {
+        let mut output = String::new();
+        tokio::io::stdin()
+            .read_to_string(&mut output)
+            .context(error::StdinReadSnafu)
+            .await?;
+        return Ok(output);
+    }
+
+    // Handle base64: URIs by decoding them directly
+    if let Some(base64_data) = input_source.strip_prefix("base64:") {
+        let decoded_bytes = engine::general_purpose::STANDARD
+            .decode(base64_data.as_bytes())
+            .context(error::Base64DecodeSnafu {
+                input_source: &input_source,
+            })?;
+
+        let decoded_string = String::from_utf8(decoded_bytes).context(error::Base64Utf8Snafu {
+            input_source: &input_source,
+        })?;
+
+        return Ok(decoded_string);
+    }
+
+    // Otherwise, the input should be a URI; parse it to know what kind.
+    let uri = Url::parse(&input_source).context(error::UriSnafu {
+        input_source: &input_source,
+    })?;
+
+    if uri.scheme() == "file" {
+        // Turn the URI to a file path, and return the file content.
+        let path = uri.to_file_path().ok().context(error::FileUriSnafu {
+            input_source: &input_source,
+        })?;
+        tokio::fs::read_to_string(path)
+            .context(error::FileReadSnafu { input_source })
+            .await
+    } else {
+        // Return the text content of the (non-file) URI.
+        let response = reqwest::get(uri).await.context(error::ReqwestSnafu {
+            uri: &input_source,
+            method: "GET",
+        })?;
+
+        let response = response.error_for_status().context(error::ReqwestSnafu {
+            uri: &input_source,
+            method: "GET",
+        })?;
+
+        response.text().await.context(error::ReqwestSnafu {
+            uri: input_source,
+            method: "GET",
+        })
```

Comment:
You may want to have a different error for this, as it is passed the point of a failed `GET` request and it is at the point of parsing the text that was returned by the request.